src/server/render.js

```diff
@@ -1,16 +1,28 @@
-import RenderStream from './render-stream'
 import { renderStartingTag } from './render-starting-tag'
 
-export function createStreamingRenderer (modules, directives, isUnaryTag) {
-  function renderComponent (component, write, next, isRoot) {
-    component.$mount()
-    renderNode(component._vnode, write, next, isRoot)
-  }
-
+export function render (modules, directives, isUnaryTag) {
   function renderNode (node, write, next, isRoot) {
     if (node.componentOptions) {
-      node.data.hook.init(node)
-      renderComponent(node.child, write, next, isRoot)
+      const { Ctor, propsData, listeners, parent, children } = node.componentOptions
+      const options = {
+        parent,
+        propsData,
+        _parentVnode: node,
+        _parentListeners: listeners,
+        _renderChildren: children
+      }
+      // check inline-template render functions
+      const inlineTemplate = node.data.inlineTemplate
+      if (inlineTemplate) {
+        options.render = inlineTemplate.render
+        options.staticRenderFns = inlineTemplate.staticRenderFns
+      }
```

Comment:
Above could probably be provided by an internal function to reduce duplicated code.